src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs

```diff
@@ -1576,13 +1577,14 @@ private Framing DetectFraming(byte[] bytes, int length)
 
             int version = -1;
 
-            if ((bytes == null || bytes.Length == 0))
+            if ((bytes == null || bytes.Length <= 0))
```

Comment:
Since the `Length` property of a .NET array is read-only (i.e. only 'get'), there is no way for the length to be negative since the contract for .NET array is supposed to prevent it.  If the `Length` property could be 'set' by a caller, I can see the value in Assert'ing that it is not negative.

So, from an object-oriented perspective, only the tests for .NET array objects themselves should be testing to prevent negative Length's.  That means that our Assert's shouldn't have to do that.